R/derive_param_framingham.R

```diff
@@ -0,0 +1,370 @@
+#'   Smoker=.52873 if smoker, 0 otherwise;
+#'   Diabetes Present=.69154 is present, 0 otherwise;
+#'   Avg Risk = 26.1931;
+#'   RiskPeriodFactor = 0.95012;
+#'
+#' For men:
+#'   Age Factor = 3.06117;
+#'   Total Chol Factor = 1.12370;
+#'   HDL Chol Factor = -0.93263;
+#'   SysBPFactor=1.93303 if not on hypertension medications, 2.99881 if on
+#'     hypertensino medications,;
+#'   Smoker=.65451 if smoker, 0 otherwise;
+#'   Diabetes Present=.57367 is present, 0 otherwise,;
+#'   Avg Risk = 23.9802;
+#'   Risk Period Factor = 0.88936;
+#'
+#' Equation:
+#' \deqn{RiskFactors = (log(Age) * AgeFactor)
+#' + (log(TotalChol) * TotalCholFactor)
+#' + (log(CholHDL) * CholHDLFactor)
+#' + (log(SysBP) * SysBPFactor) + Smoker
+#' + Diabetes Present - AvgRisk}
+#'
+#' \deqn{Risk = 100 * (1 - RiskPeriodFactor ^ exp(RiskFactors))}
+#'
+#' @author Alice Ehmann
+#'
+#' @return The input dataset with the new parameter added
+#'
+#' @keywords derivation
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' adcvrisk <- tibble::tribble(
+#'   ~USUBJID, ~PARAMCD, ~PARAM, ~AVAL, ~AVALU,
+#'   ~VISIT, ~AGE, ~SEX, ~SMOKEFL, ~DIABETFL, ~TRTHYPFL,
+#'   "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,
+#'   "mmHg", "BASELINE", 44, "F", "N", "N", "N",
+#'   "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 115,
+#'   "mmHg", "WEEK 2", 44, "F", "N", "N", "Y",
+#'   "01-701-1015", "CHOL", "Total Cholesterol (mg/dL)", 216.16,
+#'   "mg/dL", "BASELINE", 44, "F", "N", "N", "N",
+#'   "01-701-1015", "CHOL", "Total Cholesterol (mg/dL)", 210.78,
+#'   "mg/dL", "WEEK 2", 44, "F", "N", "N", "Y",
+#'   "01-701-1015", "CHOLHDL", "Cholesterol/HDL-Cholesterol (mg/dL)", 54.91,
+#'   "mg/dL", "BASELINE", 44, "F", "N", "N", "N",
+#'   "01-701-1015", "CHOLHDL", "Cholesterol/HDL-Cholesterol (mg/dL)", 26.72,
+#'   "mg/dL", "WEEK 2", 44, "F", "N", "N", "Y",
+#'   "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", 119,
+#'   "mmHg", "BASELINE", 55, "M", "Y", "Y", "Y",
+#'   "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", 101,
+#'   "mmHg", "WEEK 2", 55, "M", "Y", "Y", "Y",
+#'   "01-701-1028", "CHOL", "Total Cholesterol (mg/dL)", 292.01,
+#'   "mg/dL", "BASELINE", 55, "M", "Y", "Y", "Y",
+#'   "01-701-1028", "CHOL", "Total Cholesterol (mg/dL)", 246.73,
+#'   "mg/dL", "WEEK 2", 55,  "M", "Y", "Y", "Y",
+#'   "01-701-1028", "CHOLHDL", "Cholesterol/HDL-Cholesterol (mg/dL)", 65.55,
+#'   "mg/dL", "BASELINE", 55, "M", "Y", "Y", "Y",
+#'   "01-701-1028", "CHOLHDL", "Cholesterol/HDL-Cholesterol (mg/dL)", 44.62,
+#'   "mg/dL", "WEEK 2", 55, "M", "Y", "Y", "Y"
+#' )
+#'
+#'
+#' adcvrisk %>%
+#'   derive_param_framingham(
+#'     by_vars = vars(USUBJID, VISIT),
+#'     set_values_to = vars(
+#'       PARAMCD = "FCVD101",
+#'       PARAM = "FCVD1-Framingham CVD 10-Year Risk Score (%)"
+#'     ),
+#'     get_unit_expr = AVALU
+#'   )
+#'
+#' derive_param_framingham(
+#'   adcvrisk,
```

Comment:
`AVALU` should be removed. Otherwise it is confusing that the unit variable is not used and not populated for the new observations.